lib/browser/ipc-main-internal-utils.ts

```diff
@@ -29,9 +29,17 @@ let nextId = 0
 export function invokeInWebContents<T> (sender: Electron.WebContentsInternal, command: string, ...args: any[]) {
   return new Promise<T>((resolve, reject) => {
     const requestId = ++nextId
-    ipcMainInternal.once(`${command}_RESPONSE_${requestId}`, (
-      _event, error: Electron.SerializedError, result: any
-    ) => {
+    const channel = `${command}_RESPONSE_${requestId}`
+    ipcMainInternal.on(channel, function handler (
+      event, error: Electron.SerializedError, result: any
+    ) {
+      if (event.sender !== sender) {
```

Comment:
Can we make this check more robust to potentially having multiple instances of a single `webContents` in JS land.

I.e. Check that neither `event.sender` or `sender` are destroyed and then check the `id` properties for equality 👍

Reply:
@deepak1556, @zcbenz is this situation possible?

Reply:
The `api::WebContents` (and its JS object) is bound to `content::WebContent`, they have a 1-to-1 relationship because it is commonly needed to get the JS object from a WebContents, so by design it is impossible to have multiple JS instances for one C++ WebContents .

Reply:
so the simple comparison is enough then, right?

Reply:
@MarshallOfSound I would suggest keeping the simple comparison `event.sender != sender` to keep the code simple

Reply:
Yeah simple comparison is enough.